bin/jsii-release-golang

```diff
@@ -56,68 +72,121 @@ if [ -z "${GITHUB_TOKEN:-}" ]; then
   error "GITHUB_TOKEN env variable is required"
 fi
 
-go_mod=""
-versions=("")
-modules=("")
-
-# make sure each directory has a go.mod file
-for subdir in $(find ${dir} -mindepth 1 -maxdepth 1 -type d)
-do
-  if [ ! -f ${subdir}/go.mod ]; then
-    error "expected to find go.mod in ${subdir}. make sure you specify the root directory of all your modules."
-  fi
-  # pick one mod file to derive repository
-  go_mod=${subdir}/go.mod
+# Creates the default commit message. if the repository uses a single version
+# the message will be 'chore(release): ${VERSION}'. Otherwise, this will concatenate
+# modules with their correponding version. For example: `chore(release): module1@1.20.1 module2@1.21.1`
+function create_release_message {
+  modules=$1
 
-  # collect module names since we need them for tag prefixes.
-  modules=(${modules[@]} $(basename ${subdir}))
+  message="chore(release):"
 
-  # collect versions if they exist. used both for validation and for default value.
-  if [ -f ${subdir}/version ]; then
-    version=$(cat ${subdir}/version)
-    versions=(${versions[@]} ${version})
+  if [ ! -z ${VERSION} ]; then
+    message="${message} ${VERSION}"
+  else
+    for module in "${modules[@]}"; do
+      module_name=$(basename ${module})
+      module_version=$(extract_version ${module})
+      message="${message} ${module_name}@${module_version}"
+    done
   fi
 
-done
+  echo ${message}
+}
+
+# Extrats the version of a specific module. If `VERSION` is set, just returns that.
+# Otherwise, reads the `version` file in the module directory.
+# Fails if none exist.
+function extract_version {
+  module=$1
+  if [ ! -z "${VERSION:-}" ]; then
+    module_version=${VERSION}
+  elif [ -f ${module}/version ]; then
+    module_version=$(cat ${module}/version)
+  else
+    error "Unable to determine version of module ${module}. Either include a 'version' file, or specify a global version using the VERSION environment variable."
+  fi
 
-if [ ${go_mod} = "" ]; then
-  error "No go modules found in ${dir}"
-fi
+  echo ${module_version}
+}
+
+# Extracts the repository name from the collection of modules
+# by parsing the go.mod file of one of the modules, assuming all modules are pushed
+# to the same repository.
+function extract_repo {
+  modules=$1
+  go_mod=${modules[0]}/go.mod
+  full_module_name=$(grep module ${go_mod} | cut -d ' ' -f 2)
+  owner=$(echo ${full_module_name} | cut -d '/' -f 2)
+  repo=$(echo ${full_module_name} | cut -d '/' -f 3)
+  echo ${owner}/${repo}
```

Comment:
Would have been nice to make sure all modules use the same repo.

(should we just write this in JavaScript?)

Reply:
Done